### PR TITLE
Extend selection block using shift

### PR DESCRIPTION
**Pull request by bbrto21**
* Remove unused variable
* Change editing state first
* Enhance readability

Signed-off-by: Boram Bae <boram21.bae@samsung.com>

### CODE REVIEW

**Comment by bwikbs** [comment]
Behavior was confirmed in rpi4(gallery). It's pretty cool! thanks! :+1:

**Review by bwikbs — APPROVED** [review]

File: shell/platform/tizen/channels/text_input_channel.cc [hunk 1]
```diff
@@ -310,48 +311,58 @@ void TextInputChannel::HandleUnfilteredEvent(Ecore_Event_Key* event) {
...
+  } else if (key == "BackSpace") {
+    need_update = active_model_->Backspace();
+  } else if (key == "Delete") {
+    need_update = active_model_->Delete();
   } else if (event->string && strlen(event->string) == 1 &&
              IsASCIIPrintableKey(event->string[0])) {
     active_model_->AddCodePoint(event->string[0]);
-    SendStateUpdate(*active_model_);
+    need_update = true;
+  } else if (!strcmp(event->key, "Return") ||
```
Comment (swift-kim) [inline shell/platform/tizen/channels/text_input_channel.cc:355]:
```suggestion
  } else if (key == "Return" ||
```

File: shell/platform/tizen/channels/text_input_channel.cc [hunk 2]
```diff
@@ -310,48 +311,58 @@ void TextInputChannel::HandleUnfilteredEvent(Ecore_Event_Key* event) {
...
+             (select && !text_editing_context_.is_in_select_mode_)) {
+    EnterPressed(active_model_.get(), select);
+    return;
   } else {
-    is_filtered = false;
+    FT_LOG(Warn) << "Key[" << key << "] is unhandled!";
```
Comment (swift-kim) [inline shell/platform/tizen/channels/text_input_channel.cc:360]: Please don't use exclamation marks when it's not really important.

File: shell/platform/tizen/channels/text_input_channel.cc [hunk 3]
```diff
@@ -310,48 +311,58 @@ void TextInputChannel::HandleUnfilteredEvent(Ecore_Event_Key* event) {
...
-  } else if (!strcmp(event->key, "Right")) {
-    if (active_model_->MoveCursorForward()) {
-      SendStateUpdate(*active_model_);
-    }
-  } else if (!strcmp(event->key, "End")) {
-    if (active_model_) {
-      active_model_->MoveCursorToEnd();
-      SendStateUpdate(*active_model_);
+  bool shift = event->modifiers & ECORE_SHIFT;
+  bool need_update = false;
```
Comment (swift-kim) [inline shell/platform/tizen/channels/text_input_channel.cc:316]: nit:
```suggestion
  bool needs_update = false;
```

File: shell/platform/tizen/channels/text_input_channel.cc (hunk 1 above)
Comment (bbrto21) [reply shell/platform/tizen/channels/text_input_channel.cc]: Thanks! my mistake!

File: shell/platform/tizen/channels/text_input_channel.cc (hunk 3 above)
Comment (bbrto21) [reply shell/platform/tizen/channels/text_input_channel.cc]: Done.

File: shell/platform/tizen/channels/text_input_channel.cc (hunk 2 above)
Comment (bbrto21) [reply shell/platform/tizen/channels/text_input_channel.cc]: Ok, I removed it

**Review by swift-kim — APPROVED** [review]

**Pull request merged**